chapters/ch45.md

```diff
@@ -1,38 +1,35 @@
 
---> _replace THIS LINE by your translation for the above line_
+Hãy để xem xét một ví dụ nữa. Giả sử bạn đang xây dựng một hệ thống dịch máy từ tiếng Trung sang tiếng Anh. Hệ thống của bạn hoạt động bằng cách nhập một câu tiếng Trung *C* và tính toán một số Điểm<sub>C</sub>(*E*) cho mỗi bản dịch có thể E. Ví dụ: bạn có thể sử dụng Điểm<sub>C</sub>(*E*) = P(*E*|*C*), xác suất dịch được E cho rằng câu đầu vào là *C*.
 
 > Your algorithm translates sentences by trying to compute:
 
---> _replace THIS LINE by your translation for the above line_
+Thuật toán của bạn dịch các câu bằng cách cố gắng tính toán:
 
 ![img](../imgs/C45_01.png)
 
 > However, the set of all possible English sentences ​*E* i​ s too large, so you rely on a heuristic search algorithm.
 
---> _replace THIS LINE by your translation for the above line_
+Tuy nhiên, tập hợp các câu tiếng Anh có thể *E* quá lớn nên bạn dựa vào thuật toán tìm kiếm thực nghiệm.
 
 > Suppose your algorithm outputs an incorrect translation ​*E​*<sub>out</sub>​ rather than some correct translation ​E​*. Then the Optimization Verification test would ask you to compute whether Score​<sub>C</sub>​(*E**) > Score​<sub>C</sub>​(*E*<sub>out</sub>). If this inequality holds, then the Score​<sub>C</sub>​(.) correctly recognized E* as a superior output to *E​*<sub>out</sub>​; thus, you would attribute this error to the approximate search algorithm. Otherwise, you attribute this error to the computation of Score​<sub>C</sub>​(.).
 
---> _replace THIS LINE by your translation for the above line_
+Giả sử thuật toán của bạn dịch ra một bản không chính xác *E*<sub>out</sub> thay vì một số bản dịch chính xác E*. Sau đó, bài kiểm tra Xác minh Tối ưu hóa sẽ yêu cầu bạn tính toán xem liệu Điểm<sub>C</sub>(*E**) > Điểm<sub>C</sub>(*E*<sub>out</sub>) không. Nếu bất bình đẳng này đúng thì Điểm<sub>C</sub>(.) đã nhận dạng chính xác E* là đầu ra vượt trội hơn so với *E*<sub>out</sub>; do đó, bạn sẽ gán lỗi này cho thuật toán tìm kiếm gần đúng. Ngược lại, bạn gán lỗi này cho tính toán Điểm<sub>C</sub>(.)
 
-> It is a very common "design pattern" in AI to first learn an approximate scoring function
-> Score​<sub>x</sub>​(.), then use an approximate maximization algorithm. If you are able to spot this pattern, you will be able to use the Optimization Verification test to understand your source of errors.
+> It is a very common "design pattern" in AI to first learn an approximate scoring function Score​<sub>x</sub>​(.), then use an approximate maximization algorithm. If you are able to spot this pattern, you will be able to use the Optimization Verification test to understand your source of errors.
 
---> _replace THIS LINE by your translation for the above line_
+Đây là một “mẫu thiết kế” rất phổ biến trong AI để học hàm tính điểm gần đúng Điểm<sub>x</sub>(.), rồi sử dụng thuật toán tối đa hóa gần đúng. Nếu bạn có thể phát hiện ra kiểu mẫu này, bạn sẽ có thể sử dụng thử nghiệm Xác minh Tối ưu hóa để hiểu nguồn gốc lỗi của mình.
```

Comment:
Từ "to" trong trường hợp này không có nghĩa là "để" .

> Đây là một “mẫu thiết kế” rất phổ biến trong AI ~~để~~ _khi_ học _một_ hàm tính điểm gần đúng Điểm<sub>x</sub>(.), ~~rồi~~ _sau đó_ sử dụng _một_ thuật toán tối đa ~~hóa gần đúng~~ _xấp xỉ_.

* Cụm "spot this pattern" mình chưa nghĩ ra cách dịch hợp lý, @ngcthuong @william-vu có đề xuất gì không?

Reply:
mình ghĩ spot ở đây cũng có nghìa là tìm ra phát hiện ra, hay "tìm thấy" có vẻ chuẩn hơn.

Reply:
> mình ghĩ spot ở đây cũng có nghìa là tìm ra phát hiện ra, hay "tìm thấy" có vẻ chuẩn hơn.

Cả cụm "spot the pattern" thì dịch như thế nào nhỉ?

Reply:
@ngcthuong ^